### DISCUSSION TITLE
Renovate trying to update to old version as "latest"

### DISCUSSION THREAD

**Discussion by MatthewSH** [renovatebot/renovate#22119] [Request Help] [answered]
### I'm using the Mend Renovate hosted app on github.com.

- [X] I'm using the Mend Renovate hosted app.

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe your problem.

I'm using Renovate on a private repository in a monorepo. For the most part it works fine, except for one dependency. Trying to figure out what's going on and why it keeps pointing to a version 3 months old. Is this a me issue or Renovate?

I'm using a library called [Discordeno](https://github.com/discordeno/discordeno), specifically the dependency [@discordeno/bot](https://www.npmjs.com/package/@discordeno/bot). I'm using their latest/next version, v19. This usually comes with daily updates to the version tagging with the commit.

Right now the dependency is on `19.0.0-next.c675b45` but today they published `19.0.0-next.169e117` which is tagged as latest, even on the [NPM registry API](https://registry.npmjs.org/@discordeno/bot):

```json
"dist-tags": {
  "latest": "19.0.0-next.169e117",
  "next": "19.0.0-next.4451608"
},
```

However, Renovate keeps trying to push me back to a version from almost 3 months ago. I just added Renovate about a week ago and I've never been on this previous version as I just started using this next branch (v19) about a week and a half ago. There should be no logical reason Renovate is suggesting me this old release.

This is from the Renovate dashboard on the latest run.

```json
{
  "currentValue": "19.0.0-next.c675b45",
  "currentVersion": "19.0.0-next.c675b45",
  "datasource": "npm",
  "depName": "@discordeno/bot",
  "depType": "dependencies",
  "fixedVersion": "19.0.0-next.c675b45",
  "isSingleVersion": true,
  "lockedVersion": null,
  "packageName": "@discordeno/bot",
  "prettyDepType": "dependency",
  "registryUrl": "https://registry.npmjs.org",
  "sourceUrl": "https://github.com/discordeno/discordeno",
  "versioning": "npm",
  "warnings": [],
  "updates": [
    {
      "bucket": "non-major",
      "newVersion": "19.0.0-next.ff6dcb1",
      "newValue": "19.0.0-next.ff6dcb1",
      "releaseTimestamp": "2023-02-13T19:13:53.092Z",
      "newMajor": 19,
      "newMinor": 0,
      "updateType": "patch",
      "branchName": "renovate/discordeno-bot-19.x"
    }
  ]
}
```

For sake of showing everything, my current Renovate config.

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:js-app"
  ],
  "schedule": [
    "before 11am"
  ],
  "rollbackPrs": true,
  "dependencyDashboard": true,
  "lockFileMaintenance": {
    "enabled": true,
    "automerge": true
  },
  "minor": {
    "automerge": true
  },
  "patch": {
    "automerge": true
  },
  "pin": {
    "automerge": true
  },
  "digest": {
    "automerge": true
  }
}
```

**Comment by rarkins** [comment, marked as answer]
The problem is that this package is not following semantic versioning with its "next" versions. The use of commit SHAs do not _sort_. You most likely need to configure followTag=latest and just stay pinned to latest.

Renovate sees you're on `19.0.0-next.c675b45` but latest is `19.0.0-next.169e117` so thinks "this repo is already past latest, so I guess it doesn't care about the tag and wants to be updated to the highest sorting version"

**Reply by MatthewSH** [reply]
I'm relatively new to Renovate so take what I say with a grain of salt.

That feels kind of different to the intent of the application, no? If the intent is to maintain to the latest version, the latest version tag on NPM will always be the latest? Unless I'm mistaken and Renovate does more logic to keep updating patch versions on a package that is not marked as latest? For example, v2 of a package is the "latest" but they still release patches for v1.

I added it, but it's treating it as a rollback which makes sense if it's doing an alphabetical sort:
![image](https://github.com/renovatebot/renovate/assets/3768988/05355ff6-7b6b-4a67-851f-0b9d5c4372ee)

It *feels* like the application should recognize that there's a commit hash and be able to update based on that, but that may just be a pipe dream.

**Reply by rarkins** [reply]
> That feels kind of different to the intent of the application, no? If the intent is to maintain to the latest version, 

This is your misunderstanding. Keeping up to latest is just one, common use of Renovate. Others use it to keep updated to unstable/canary versions too.

You also should understand that the way this package is doing versioning is not in the spirit of semver because versions do not sort in the order they were published

**Reply by MatthewSH** [reply]
Fair enough! Thanks for correcting me there @rarkins! I appreciate the work you all do and thanks for the answer.
